packages/fx-core/src/plugins/resource/identity/utils/telemetryUtil.ts

```diff
@@ -42,14 +39,17 @@ export class TelemetryUtils {
     properties[Telemetry.properties.errorCode] = errorCode;
     properties[Telemetry.properties.errorType] = errorType;
     properties[Telemetry.properties.errorMessage] = errorMessage;
-    properties[Telemetry.properties.component] = Telemetry.componentName;
-    if (this.ctx.app.id) {
-      properties[Telemetry.properties.appid] = this.ctx.app.id;
-    }
+    this.addProperties(properties);
     TelemetryUtils.ctx.telemetryReporter?.sendTelemetryErrorEvent(
       eventName,
       properties,
       measurements
     );
   }
+
+  private static addProperties(properties: { [key: string]: string; }) {
+    properties[Telemetry.properties.component] = Telemetry.componentName;
+    const appId = this.ctx.configOfOtherPlugins.get(Constants.solution)?.get(Constants.remoteTeamsAppId) as string;
```

Comment:
If remote teams app id is not available, here will be "undefined as string", which I remember will throw error?

Reply:
I debugged it locally, and it will not trigger an error.

Reply:
node env should throw error.....

Reply:
The value type of config map is like string | number | undefined, I tested it and it was safe. But I didn't find docs to support this casting. I will fix it.